Updated DCF end-user UX and extension request details (#81459)
Azure/azure-content: articles/active-directory/external-identities/google-federation.md

- Microsoft 365 services that are accessed through a website (for example, SharePoint Online, Office web apps, and Teams web app)
- Mobile apps using system web-views for authentication ([SFSafariViewController](https://developer.apple.com/documentation/safariservices/sfsafariviewcontroller) on iOS, [Custom Tabs](https://developer.chrome.com/docs/android/custom-tabs/overview/) on Android).  
- Google Workspace identities, for example when you’re using [SAML-based federation](direct-federation.md) with Google Workspace
- Windows apps that use the Web Account Manager (WAM) or Web Authentication Broker (WAB).  

### Action needed for embedded web-views

Modify your apps to use the system browser for sign-in. For details, see [Embedded vs System Web UI](../develop/msal-net-web-browsers.md#embedded-vs-system-web-ui) in the MSAL.NET documentation. All MSAL SDKs use the system web-view by default.

### What to expect

> Globally rolling out starting September 30, 2021, Microsoft will deploy device login flow that serves as a workaround for apps still using embedded web-views to ensure that authentication isn't blocked. 

Reviewer: ```suggestion
Starting September 30, Microsoft will globally roll out a device sign-in flow that serves as a workaround for apps still using embedded web-views to ensure that authentication isn’t blocked.
```

> ### How to sign in with device login flow

Reviewer: ```suggestion
### How to sign in with the device sign-in flow
```

> The device login flow will prompt users signing in with a Gmail account in an embedded web-view to enter a code in a separate browser to finish signing in. The following is what users can expect to see if they are signing in with their Gmail account for the first time with no active sessions in the browser. If there is an existing Gmail already signed in, some steps may be eliminated.

Reviewer: ```suggestion
The device sign-in flow prompts users who sign in with a Gmail account in an embedded web-view to enter a code in a separate browser before they can finish signing in. If users are signing in with their Gmail account for the first time with no active sessions in the browser, they'll see the following sequence of screens. If an existing Gmail account is already signed in, some of these steps might be eliminated.
```

> 1. On the Sign in screen, users have the option to enter their Gmail then click "Next," or go to "Sign-in options" then select "Sign in from another device."

Reviewer: ```suggestion
1. On the **Sign in** screen, users can enter their Gmail address and then select **Next**, or they can select **Sign-in options** and choose **Sign in from another device**.
```

   ![image](https://user-images.githubusercontent.com/15203472/134457320-7d1e5636-f163-4a51-ab0f-5fae168a0a5c.png)
   

> 2. They will see the following screen that prompts them to open a new window to navigate to https://microsoft.com/devicelogin and enter the 9-digit alphanumeric code displayed on the screen.

Reviewer: ```suggestion
2. The following screen appears, prompting the user to open a new window, navigate to https://microsoft.com/devicelogin, and enter the 9-digit alphanumeric code displayed.
```

   ![image](https://user-images.githubusercontent.com/15203472/134458974-8ea88a64-f45e-4801-b82a-86ac9fcc6bd8.png)

> 3. The device login page will prompt the user to enter code. 

Reviewer: ```suggestion
3. The device sign-in page opens, where the user can enter the code. 
```

   ![image](https://user-images.githubusercontent.com/15203472/134457819-b2f0706c-8493-4a3d-9437-577a5420dfd2.png)
   

> 4. If the codes match, for security purposes, the user will be asked to enter their email again to confirm they are signing in to the app and location displayed.

Reviewer: ```suggestion
4. If the codes match, for security purposes the user is asked to reenter their email to confirm their app and sign-in location.
```

   ![image](https://user-images.githubusercontent.com/15203472/134458107-0ab2496f-9956-4b97-9a99-77d5b3389721.png)
   

> 5. The user will sign in through Google by entering email and password.

Reviewer: ```suggestion
5. The user signs in to Google with their email and password.
```

   ![image](https://user-images.githubusercontent.com/15203472/134458704-7399bb72-8dfb-411d-9690-3e1932f35bc2.png)
   

> 7. Once again, they will be asked to confirm the application they're signing into.

Reviewer: ```suggestion
7. Once again, they're asked to confirm the application they're signing into.
```

   ![image](https://user-images.githubusercontent.com/15203472/134458470-b51c5892-4eeb-4e21-a512-07d2d291b80b.png)

> 7. If confirmed, the user will be successfully signed in and asked to close the tab or window to go back to the first screen where the user is now signed into the app.

Reviewer: ```suggestion
7. The user selects **Continue**. A prompt confirms that they’re signed in. The user closes the tab or window and is returned to the first screen, where they're now signed in to the app.
```

   ![image](https://user-images.githubusercontent.com/15203472/134458364-659c8835-2895-41a3-86af-2ed63f5fcfa7.png)

Alternatively, you can have your existing and new Gmail users sign in with email one-time passcode. To have your Gmail users use email one-time passcode:
1. [Enable email one-time passcode](one-time-passcode.md#enable-email-one-time-passcode)
2. [Remove Google Federation](google-federation.md#how-do-i-remove-google-federation)
3. [Reset redemption status](reset-redemption-status.md) of your Gmail users so they can use email one-time passcode going forward.

> If you wish to request extension, impacted customers with affected OAuth client ID(s) should have received an email from Google Developers with the following information regarding a one-time policy enforcement extension that needs to be filled out by Jan 31, 2022:

Reviewer: ```suggestion
If you want to request extension, impacted customers with affected OAuth client ID(s) should have received an email from Google Developers with the following information regarding a one-time policy enforcement extension, which must be completed by Jan 31, 2022:
```

> >If necessary, you may request a one-time [policy enforcement extension for embedded webviews](https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fnotifications.google.com%2Fg%2Fp%2FAD-FnEz_hbffaJlsX17rxSVBgq0aEQtIZGZdcC8wspVMZ_TP939PQLp-J-DQekKQXV4xybzyGqpZZG0tzVXwzfdwwQJVX4wT9bBMTLdAt-s0gpn142uWYRhs8Ym07tlbNKd5hYE8UAxftcGW-eIM7Rbt9QYUNMU1ghtiHAyFCh6_qVc8jkiQFxADCEMDYYPsLjcCmnErQCvJRrbXn0SVPowayOIQCT5EXjJ6LZfm&data=04%7C01%7CLinda.Park%40microsoft.com%7C16025a5cd67b494a544908d97a3f60e6%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C637675234894428918%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=mnxmd9v5nOGOqnGkVw6Dn%2FxKESKFAbTmoznFqL7K%2BjI%3D&reserved=0) for each listed OAuth client ID until January 31, 2022. For clarity, the policy for embedded webviews will be enforced on February 1, 2022 with no exceptions or extensions.

Reviewer: ```suggestion
   >If necessary, you may request a one-time **policy enforcement extension for embedded webviews** for each listed OAuth client ID until January 31, 2022. For clarity, the policy for embedded webviews will be enforced on February 1, 2022 with no exceptions or extensions.
```
